### ISSUE TITLE
RA gets confused if field name coincides with function parameters

### ISSUE THREAD

**Issue by SchrodingerZhu** [rust-lang/rust-analyzer#15034]
```rust
#[derive(Eq, PartialEq, Hash)]
struct A {
    state: usize,
}
```
which expands to
```rust
 #[automatically_derived]
    impl ::core::hash::Hash for A {
        fn hash<__H: ::core::hash::Hasher>(&self, state: &mut __H) -> () {
            ::core::hash::Hash::hash(&self.state, state)
        }
    }
```
Here `state` is a function parameter for `Hash::hash`. RA somehow gets confused by this similarity and generates a false positive.

<img width="1440" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/20108837/39f1b418-7d92-4d4b-b3aa-caaf7a160ce4">

**rust-analyzer version**: 0.4.1548-standalone (d567091f4 2023-06-11)

**rustc version**: rustc 1.71.0-nightly (74c482104 2023-05-04)

**relevant settings**: n/a

**Comment by SchrodingerZhu** [comment]
Glad to know this is captured.
